torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -537,10 +537,20 @@ void ProcessGroupNCCL::WorkNCCL::synchronizeInternal(
   // Device synchronize only after we've completed timeout checks.
   if (!barrierTensors_.empty()) {
     // If we use the work to do barrier, we should block here
-    at::cuda::OptionalCUDAGuard gpuGuard;
     for (auto& device : devices_) {
-      gpuGuard.set_index(device.index());
-      AT_CUDA_CHECK(cudaDeviceSynchronize());
+      // `dist.barrier()` only requires all CPU processes to enter this
```

Comment:
We should take this chance to update the documentation of dist::barrier to cover NCCL and explain the expected behavior around CPU and accelerators.